x-pack/plugins/maps/public/kibana_services.d.ts

```diff
@@ -4,6 +4,8 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { IIndexPattern, DataPublicPluginStart } from 'src/plugins/data/public';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { IndexPatternsService } from '../../../../src/plugins/data/common/index_patterns/index_patterns/index_patterns';
```

Comment:
This is available from `src/plugins/data/public/index_patterns/index.ts` if you want to get rid of the eslint comment

Reply:
Thanks, I will use this import path. eslint comment is still needed because have to go into public folder buts this is still better.